src/HUD.cpp

```diff
@@ -0,0 +1,203 @@
+#include "HUD.h"
+#include "common.h"
+
+
+osg::Camera* createHUD(const std::string& logoFile, float scale, int winWidth,
+                       int winHeight)
+{
+    // --- HUD Camera ---
+    osg::Camera* hudCamera = new osg::Camera;
+
+    hudCamera->setReferenceFrame(osg::Transform::ABSOLUTE_RF);
+    hudCamera->setClearMask(GL_DEPTH_BUFFER_BIT);
+    hudCamera->setRenderOrder(osg::Camera::POST_RENDER);
+    hudCamera->setProjectionMatrix(
+        osg::Matrix::ortho2D(0, winWidth, 0, winHeight));

```

Comment:
Your approach works fine as long as we assume that the window’s width and height remain constant, which is not guaranteed. After resizing the window, the original aspect ratio will be lost, leading to distortions like the one shown in the picture. I suggest keeping a handle to the `hudCamera `(e.g., near the `viewer `in `common.h`) and updating its settings accordingly whenever the window size changes. This will ensure the HUD maintains the correct aspect ratio.


![screen_shot_0_0](https://github.com/user-attachments/assets/22ffd978-ff5e-44eb-9f71-c209f807905e)